shared/src/Twin.cs

```diff
@@ -151,6 +151,19 @@ public Twin(TwinProperties twinProperties)
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
         public X509Thumbprint X509Thumbprint { get; internal set; }
 
+        /// <summary>
+        /// The scope of the device. Auto-generated and immutable for edge devices and modifiable in leaf devices to create child/parent relationship.
+        /// </summary>
+        [DefaultValue(null)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public string DeviceScope { get; internal set; }
```

Comment:
I noticed these values are available in the twin too, so I added them here. However, note this property is read-only and `ParentScopes` I used `IReadOnlyList`. This effectively makes any values deserialized as read-only, as my experimentation showed that changing the values here had no affect.

I chose to make the values visible here so if someone had the twin (since using `RegistryManager` they can call `Query()` to get twins but cannot enumerate devices. If someone just wanted to draw a picture of device hierarchy, they should be able to do that just by enumerating twins, and not having to call `GetDeviceAsync()` on each device Id.